### PR TITLE
Create `user_joined_channel` event triggers for welcome messages

### LINKED ISSUE #10: Update Welcome Bot to include an Event Trigger
References https://slack-pde.slack.com/archives/C03PJAXUGK0/p1667497994694289

Example in another repository: https://slack-github.com/slack-services/platform-2.0-welcome-bot-lab/blob/master/triggers/welcome_user_event.ts

### PR DESCRIPTION
**Pull request by e-zim**
### Type of change

- [ ] New sample
- [x] New feature
- [x] Bug fix
- [ ] Documentation

### Summary

This PR creates a `user_joined_channel` event trigger for a channel (if none currently exists) after a welcome message is created. When this trigger is tripped, the stored welcome message will be sent ephemerally to the joining user. Closes #10.

### Requirements

- [x] I’ve checked my submission against the Samples Checklist to ensure it complies with all standards
- [x] I have ensured the changes I am contributing align with existing patterns and have tested and linted my code
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct)

### CODE REVIEW

File: functions/create_welcome_message.ts [hunk 1]
```diff
@@ -28,34 +30,104 @@ export const WelcomeMessageSetupFunction = DefineFunction({
...
+  // Return if any matching triggers were found
+  const exists = joinedTriggers.length > 0;
+  return { exists };
+}
+
+/**
+ * saveUserJoinedChannelTrigger creates a new user_joined_channel trigger
+ * for the "Send Welcome Message" workflow in a channel.
+ */
+export async function saveUserJoinedChannelTrigger(
```
Comment (misscoded) [inline functions/create_welcome_message.ts:107]: For my own understanding and lack of having taken this approach first-hand previously: if this is the act of creating a Trigger, why is this here rather than in `triggers/`?

**Review by misscoded — APPROVED** [review]
LGTM :shipit: 

I left a question regarding the decision to place trigger logic inside of the function directly, but my guess is that there's a reason we've done so here that I haven't come across myself yet.

Also, seeking confirmation that a once-over of the README has been done and no changes need to be made to reflect the update.

File: functions/create_welcome_message.ts (hunk 1 above)
Comment (e-zim) [reply functions/create_welcome_message.ts:107]: A solid question for sure. My reasoning is that this trigger is being created at runtime and requires a dynamic input (the `channel`), while the definitions found in `triggers/` usually seem to be static.

I will create a follow up PR that moves some of this logic into the `triggers/` directory for further discussion since I see a few advantages, disadvantages, and further questions raised by this approach. I think this approach is sufficient for this PR though, so will merge it as is.
